apps/condo/domains/miniapp/schema/B2CAppProperty.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * Generated by `createschema miniapp.B2CAppProperty 'app:Relationship:B2CApp:PROTECT; address:Text;' --force`
+ */
+
+const dayjs = require('dayjs')
+const get = require('lodash/get')
+const { Text, Relationship } = require('@keystonejs/fields')
+const { GQLListSchema, getById } = require('@core/keystone/schema')
+const { historical, versioned, uuided, tracked, softDeleted } = require('@core/keystone/plugins')
+const { dvAndSender } = require('@condo/domains/common/schema/plugins/dvAndSender')
+const access = require('@condo/domains/miniapp/access/B2CAppProperty')
+const { getAddressSuggestions } = require('@condo/domains/common/utils/serverSideAddressApi')
+const { VALID_HOUSE_TYPES } = require('@condo/domains/property/constants/common')
+const {
+    NO_APP_ERROR,
+    RESTRICT_PROPERTY_APP_CHANGE_ERROR,
+    INCORRECT_ADDRESS_ERROR,
+    INCORRECT_HOUSE_TYPE_ERROR,
+} = require('@condo/domains/miniapp/constants')
+
+
+const B2CAppProperty = new GQLListSchema('B2CAppProperty', {
+    schemaDoc: 'Link between specific home address and B2C App. used to filter B2C applications that can be run on a specific address',
+    labelResolver: async (item) => {
+        const app = await getById('B2CApp', item.app)
+        const appName = get(app, 'name', 'deleted')
+        const appDeveloper = get(app, 'developer', 'deleted')
+        return `${appDeveloper}-${appName}-${item.address}`
+    },
+    fields: {
+        app: {
+            schemaDoc: 'Link to B2C App',
+            type: Relationship,
+            ref: 'B2CApp.properties',
+            isRequired: false,
+            kmigratorOptions: { null: true, on_delete: 'models.PROTECT' },
+            access: {
+                update: access.canUpdateAppLink,
+            },
+            hooks: {
+                validateInput: async ({ operation, resolvedData, addFieldValidationError }) => {
+                    if (operation === 'update') {
+                        if (resolvedData.hasOwnProperty('app') && resolvedData.app !== null) {
+                            return addFieldValidationError(RESTRICT_PROPERTY_APP_CHANGE_ERROR)
```

Comment:
it' look like you already check it by access?

Reply:
was admin panel hack. removed